taipy/gui_core/_adapters.py

```diff
@@ -225,11 +227,15 @@ def get(self):
                         self.__get_data(datanode),
                         datanode._edit_in_progress,
                         datanode._editor_id,
-                        _get_reason(is_readable(datanode), "Datanode not readable"),
-                        _get_reason(is_editable(datanode), "Datanode not editable"),
+                        _get_reason(is_readable(datanode), "Data node not readable"),
+                        _get_reason(is_editable(datanode), "Data node not editable"),
+                        isinstance(datanode, _FileDataNodeMixin),
+                        "Data unavailable: The data node has never been written."
+                        if isinstance(datanode, _FileDataNodeMixin) and not datanode._get_downloadable_path()
```

Comment:
I opened an API to check if a data node file can be downloaded returning a `ReasonCollection `object: `datanode.is_downloadable()`.